overlord/snapstate/snapmgr.go

```diff
@@ -884,8 +883,13 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if cand.Revision.Local() {
+		snapst.LocalRevision = ss.Revision()
+	}
+
```

Comment:
mmh, it's correct for an install/update but not for a revert/refresh to local revision I suppose (it would make the LocalRevision go back in time)